lib/compute/agent-nodes.ts

```diff
@@ -16,59 +16,55 @@ export class AgentNodes {
     // Refer: https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/ec2/model/InstanceType.html for instance types
     readonly AL2_X64: AgentNodeProps;
 
-    readonly AL2_X64_DOCKER_1: AgentNodeProps;
+    readonly AL2_X64_DOCKER_HOST: AgentNodeProps;
 
     readonly AL2_ARM64: AgentNodeProps;
 
-    readonly AL2_ARM64_DOCKER_1: AgentNodeProps;
+    readonly AL2_ARM64_DOCKER_HOST: AgentNodeProps;
 
-    readonly UBUNTU_X64_DOCKER: AgentNodeProps;
+    readonly UBUNTU_X64_DOCKER_BUILDER: AgentNodeProps;
 
     constructor(stack: Stack) {
       this.AL2_X64 = {
-        workerLabelString: 'AL2-X64',
+        workerLabelString: 'Jenkins-Agent-al2-x64-c54xlarge-Single-Host',
         instanceType: 'C54xlarge',
         remoteUser: 'ec2-user',
-        amiId: MachineImage.latestAmazonLinux({
-          generation: AmazonLinuxGeneration.AMAZON_LINUX_2,
-          cpuType: AmazonLinuxCpuType.X86_64,
-        }).getImage(stack).imageId.toString(),
-        initScript: 'sudo mkdir -p /var/jenkins/ && sudo chown -R ec2-user:ec2-user /var/jenkins &&'
-        + ' sudo yum install -y java-1.8.0-openjdk cmake python3 python3-pip python3-devel && sudo yum groupinstall -y "Development Tools" &&'
-        + ' sudo ln -sfn `which pip3` /usr/bin/pip && pip3 install pipenv && sudo ln -s ~/.local/bin/pipenv /usr/local/bin',
+        numExecutors: 1,
+        amiId: 'ami-00a07e55fcad01043',
+        initScript: 'sudo yum update -y || sleep 10 && sudo yum update -y',
       };
-      this.AL2_X64_DOCKER_1 = {
+      this.AL2_X64_DOCKER_HOST = {
         workerLabelString: 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host',
         instanceType: 'C54xlarge',
         remoteUser: 'ec2-user',
+        numExecutors: 8,
```

Comment:
Won't 8 be too much? How about 4 to start with?

Reply:
ideally it should be half the number of cores.

Reply:
8 is not too much we have 16 core here and each use 2 core is fine, we have been using this exact same setup in jenki-jenki for nearly a year.